xbmc/input/InputManager.cpp

```diff
@@ -834,14 +834,16 @@ void CInputManager::UnregisterKeyboardHandler(KEYBOARD::IKeyboardHandler* handle
 
 void CInputManager::RegisterMouseHandler(MOUSE::IMouseInputHandler* handler)
 {
-  auto it = m_mouseHandlers.find(handler);
+  auto it = std::find_if(m_mouseHandlers.begin(), m_mouseHandlers.end(),
+    [handler](const std::pair<MOUSE::IMouseInputHandler*, std::unique_ptr<MOUSE::IMouseDriverHandler>>& element) { return element.first == handler; });
   if (it == m_mouseHandlers.end())
-    m_mouseHandlers[handler].reset(new MOUSE::CMouseInputHandling(handler, m_mouseButtonMap.get()));
+    m_mouseHandlers.emplace_back(std::make_pair(handler, std::unique_ptr<MOUSE::IMouseDriverHandler>(new MOUSE::CMouseInputHandling(handler, m_mouseButtonMap.get()))));
```

Comment:
Do we want mouse handlers to be processed in the same order as RegisterMouseHandler() is called, or should priority be given to the most-recently registered mouse handler?